routes/markets.js

```diff
@@ -1,8 +1,27 @@
 var express = require('express');
 var router = express.Router();
+var firebase = require('firebase');
+var config = require('./config.js');
 
-router.get('/', function(req, res, next) {
+if(!firebase.apps.length){
+	firebase.initializeApp(config.config);
+}
+const db = firebase.database();
+
+router.get('/', isAuthorized, function(req, res, next) {
 	res.render('markets');
 });
 
+function isAuthorized(req, res, next){
+	firebase.auth().onAuthStateChanged(function(user) {
```

Comment:
This is from the firebase docs, the way I go about authorizing a user is to see if they're signed in and the method I used is the recommended way to go about that. 

Get the currently signed-in user
The recommended way to get the current user is by setting an observer on the Auth object:

firebase.auth().onAuthStateChanged(function(user) {
  if (user) {
    // User is signed in.
  } else {
    // No user is signed in.
  }
});

By using an observer, you ensure that the Auth object isn't in an intermediate state—such as initialization—when you get the current user. When you use signInWithRedirect, the onAuthStateChanged observer waits until getRedirectResult resolves before triggering.

Reply:
Yeah but isn't that for clients not servers? Servers don't have "currently signed in users" because multiple users should be able to access the same server at the same time (unless you keep track of which users are currently signed in, but it's generally strongly discouraged to keep track of existing sessions&mdash;servers should be stateless).